x-pack/plugins/threat_intelligence/public/modules/indicators/lib/display_value.ts

```diff
@@ -41,44 +190,32 @@ const rulesArray: MapperRule[] = [
   [isIpIndicator, extractIp],
   [isUrlIndicator, extractUrl],
   [isFileIndicator, extractFile],
+  [isUnknownIndicator, extractId],
+  [isEmailAddress, extractEmailAddress],
+  [isDomain, extractDomain],
+  [isX509Certificate, extractX509Serial],
+  [isWindowsRegistryKey, extractWindowsRegistryKey],
+  [isAutonomousSystem, extractAutonomousSystemNumber],
+  [isMacAddress, extractMacAddress],
 ];
 
 /**
  * Finds display value mapping function for given indicatorType
  */
-const findMappingRule = (indicatorType: string | null): IndicatorValueExtractor => {
-  const [_, extract = extractStub] = rulesArray.find(([check]) => check(indicatorType)) || [];
+const findMappingRule = (indicatorType: string | null): IndicatorDisplayNameExtractor => {
+  const [_, extract = extractId] = rulesArray.find(([check]) => check(indicatorType)) || [];
   return extract;
 };
 
 /**
  * Cached rules for indicator types
  */
-const rules: Record<string, IndicatorValueExtractor> = {};
-
-/**
- * Mapping between the indicator type and the {@link RawIndicatorFieldId}.
- */
-const indicatorTypeToField: { [id: string]: RawIndicatorFieldId } = {
-  file: RawIndicatorFieldId.FileSha256,
-  'ipv4-addr': RawIndicatorFieldId.Ip,
-  'ipv6-addr': RawIndicatorFieldId.Ip,
-  url: RawIndicatorFieldId.UrlFull,
-};
-
-/**
- * Find and return indicator display value field
- */
-export const displayField = (indicator: Indicator): string | null => {
-  const indicatorType = (unwrapValue(indicator, RawIndicatorFieldId.Type) || '').toLowerCase();
-
-  return indicatorTypeToField[indicatorType];
-};
+const rules: Record<string, IndicatorDisplayNameExtractor> = {};
 
 /**
- * Find and return indicator display value, if possible
+ * Find and return indicator display name structure [field, value]
  */
-export const displayValue = (indicator: Indicator): string | null => {
+export const getDisplayName = (indicator: Indicator): IndicatorDisplayName => {
```

Comment:
@maxcold  should we rename the file from `display_value.ts` to `display_name.ts (and the unit test file) to match all these changes?

Reply:
makes sense! done